### PR TITLE
Clean up around opt-in banner on non-secrets-sync views

### PR DESCRIPTION
**Pull request by Monkeychip**
TODO:
- [ ] fix tests

This PR:

1. Removes the network request to GET activation-flags and the opt-in banner on OSS versions. Preview WIP.

2. Removes the network request to GET activation-flags and the opt-in banner on Enterprise versions that do _not_ have `secrets-sync` returned on the unauthenticated `license/features` endpoint.

WIP...

### CODE REVIEW

File: ui/lib/sync/addon/components/secrets/page/overview.ts [hunk 1]
```diff
@@ -18,13 +18,14 @@ import type RouterService from '@ember/routing/router-service';
 import type VersionService from 'vault/services/version';
 import type { SyncDestinationAssociationMetrics } from 'vault/vault/adapters/sync/association';
 import type SyncDestinationModel from 'vault/vault/models/sync/destination';
-import type AdapterError from '@ember/test/adapter';
+// import type AdapterError from '@ember/test/adapter';
```
Comment (hashishaw) [inline ui/lib/sync/addon/components/secrets/page/overview.ts:21]: 🧹

File: ui/lib/sync/addon/components/secrets/page/overview.ts [hunk 2]
```diff
@@ -18,13 +18,14 @@ import type RouterService from '@ember/routing/router-service';
...
 
 interface Args {
   destinations: Array<SyncDestinationModel>;
   totalVaultSecrets: number;
   activatedFeatures: Array<string>;
-  adapterError: AdapterError | null;
+  adapterError: any;
+  // adapterError: AdapterError | null;
```
Comment (hashishaw) [inline ui/lib/sync/addon/components/secrets/page/overview.ts:28]: 🧹

File: ui/lib/sync/addon/components/secrets/page/overview.ts [hunk 3]
```diff
@@ -52,6 +57,24 @@ export default class SyncSecretsDestinationsPageComponent extends Component<Args
     return this.args.activatedFeatures.includes('secrets-sync');
   }
 
+  get adapterErrorMessage() {
```
Comment (hashishaw) [inline ui/lib/sync/addon/components/secrets/page/overview.ts:60]: We won't have an adapterError from getting this endpoint anymore, we can remove!

File: ui/lib/sync/addon/components/secrets/page/overview.ts [hunk 4]
```diff
@@ -52,6 +57,24 @@ export default class SyncSecretsDestinationsPageComponent extends Component<Args
...
+        ? 'You do not have read access to the sys/activation-flags endpoint. We cannot determine if Secrets Sync has been activated or not.'
+        : this.args.adapterError.message;
+    }
+    return '';
+  }
+
+  get postErrorMessage() {
+    // make clear if this is a permissions issue
+    if (this.error) {
+      return 'Activation failed. You do not seem to have the permission to enable the feature.';
```
Comment (hashishaw) [inline ui/lib/sync/addon/components/secrets/page/overview.ts:73]: "seem to" feels casual for the brand. Suggestion below 👇 
```suggestion
      return 'Activation failed. You may not have the permission to enable the feature.';
```

File: ui/lib/sync/addon/routes/secrets.ts [hunk 5]
```diff
@@ -20,17 +21,23 @@ interface ActivationFlagsResponse {
...
-        return resp.data?.activated;
-      })
-      .catch((error: AdapterError) => {
-        return error;
-      });
+    // only fetch activated features for enterprise licenses that include the secrets-sync feature
+    if (this.version.hasSecretsSync) {
+      return await this.store
+        .adapterFor('application')
+        .ajax('/v1/sys/activation-flags', 'GET')
```
Comment (hashishaw) [inline ui/lib/sync/addon/routes/secrets.ts:31]: Another spot we'll have to merge carefully!

File: ui/tests/integration/components/sync/secrets/landing-cta-test.js
Comment (hashishaw) [inline ui/tests/integration/components/sync/secrets/landing-cta-test.js:1]: 🎉

File: ui/lib/sync/addon/components/secrets/page/overview.ts (hunk 4 above)
Comment (hellobontempo) [reply ui/lib/sync/addon/components/secrets/page/overview.ts:73]: I think we can just return the error from the backend here instead of intercepting ?

**Review by hashishaw — APPROVED** [review]
🚀

**Pull request merged**